plugins/modules/ad_auth_provider.py

```diff
@@ -232,6 +231,18 @@
 API_VERSION = "authentication/v2"
 
 
+def remove_item(result):
+    if result:
+        for i in range(len(result["servers"])):
+            if (
+                result["servers"][i]["binding"]
+                and "password" in result["servers"][i]["binding"]
+            ):
+                del result["servers"][i]["binding"]["password"]
```

Comment:
Maybe we should simplify this a bit:

```suggestion
        for server in result["servers"]:
            if server["binding"] and "password" in server["binding"]:
                del server["binding"]["password"]
```